tests/test_mailtrap_backend.py

```diff
@@ -0,0 +1,299 @@
+        self.message.track_opens = True
+        response = self.message.send()
+        self.assertEqual(response, 1)
+
+    def test_template_id(self):
+        self.message.template_id = "template_id"
+        response = self.message.send()
+        self.assertEqual(response, 1)
+        data = self.get_api_call_json()
+        self.assertEqual(data["template_uuid"], "template_id")
+
+    def test_merge_data(self):
+        self.message.merge_data = {"to@example.com": {"name": "Recipient"}}
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+    def test_merge_global_data(self):
+        self.message.merge_global_data = {"global_name": "Global Recipient"}
+        response = self.message.send()
+        self.assertEqual(response, 1)
+        data = self.get_api_call_json()
+        self.assertEqual(
+            data["template_variables"], {"global_name": "Global Recipient"}
+        )
+
+    def test_esp_extra(self):
+        self.message.esp_extra = {"custom_option": "value"}
+        response = self.message.send()
+        self.assertEqual(response, 1)
+        data = self.get_api_call_json()
+        self.assertEqual(data["custom_option"], "value")
+
+
+@tag("mailtrap")
+class MailtrapBackendRecipientsRefusedTests(MailtrapBackendMockAPITestCase):
+    """
+    Should raise AnymailRecipientsRefused when *all* recipients are rejected or invalid
+    """
+
+    @unittest.skip("TODO: is this test correct/necessary?")
+    def test_recipients_refused(self):
+        self.set_mock_response(
+            status_code=400, raw=b'{"error": "All recipients refused"}'
+        )
+        with self.assertRaises(AnymailRecipientsRefused):
+            self.message.send()
+
+    @unittest.skip(
+        "TODO: is this test correct/necessary? How to handle this in mailtrap backend?"
+    )
+    def test_fail_silently(self):
+        self.set_mock_response(
+            status_code=400, raw=b'{"error": "All recipients refused"}'
+        )
+        self.message.fail_silently = True
+        sent = self.message.send()
+        self.assertEqual(sent, 0)
+
+
+@tag("mailtrap")
+class MailtrapBackendSessionSharingTestCase(
+    SessionSharingTestCases, MailtrapBackendMockAPITestCase
+):
+    """Requests session sharing tests"""
+
+    pass  # tests are defined in SessionSharingTestCases
+
+
+@tag("mailtrap")
+@override_settings(EMAIL_BACKEND="anymail.backends.mailtrap.EmailBackend")
+class MailtrapBackendImproperlyConfiguredTests(AnymailTestMixin, SimpleTestCase):
+    """Test ESP backend without required settings in place"""
+
+    def test_missing_api_token(self):
+        with self.assertRaises(ImproperlyConfigured) as cm:
+            mail.send_mail("Subject", "Message", "from@example.com", ["to@example.com"])
+        errmsg = str(cm.exception)
+        self.assertRegex(errmsg, r"\bMAILTRAP_API_TOKEN\b")
+        self.assertRegex(errmsg, r"\bANYMAIL_MAILTRAP_API_TOKEN\b")
```

Comment:
Probably also need to test various combinations of settings, both valid and error cases (api_url, testing_enabled, test_inbox_id), to make sure the right api is called or a helpful error is raised.